robocop/checkers/naming.py

```diff
@@ -668,8 +681,18 @@ def visit_SectionHeader(self, node):  # noqa
                 end_col=node.col_offset + len(name) + 1,
             )
 
+    def visit_File(self, node):  # noqa
+        for section in node.sections:
+            if isinstance(section, TestCaseSection) and "task" in section.header.name.lower():
+                self.task_section = True
```

Comment:
What if it is translated section name? I believe that since RF 6 the Task header is actually TaskSectionHeader instead of Test Case like before. We can use it to support ltranslations since RF 6. Pre 6 your code should work fine

Reply:
I actually haven't found `TaskSectionHeader`. In the latest docs there is:
![image](https://github.com/MarketSquare/robotframework-robocop/assets/7412964/8fbe7380-1d63-42cd-ad62-d96246267a86)

Reply:
The section is TestCaseSection for both, but now the header can be TASK_HEADER (see your screenshot, the last line).

To test this theory, I have created following file:

```
language: pl


*** Zadania ***
Test Timeout    1 min
```
and put breakpoint in visit_TestCaseSection. That's what I got:

![image](https://github.com/MarketSquare/robotframework-robocop/assets/8532066/30274cbe-d99f-4d89-845c-231b309d3126)

It properly recognized that "Zadania" is Task header - so for RF 6 it's better to check for Task Header (note it doesn't exist pre RF 6)

Reply:
I've also added *** Ustawienia *** with Test/Task Timeouts to see how they look like when translated:

Task Timeout:
![image](https://github.com/MarketSquare/robotframework-robocop/assets/8532066/b4c16e01-e070-48ad-91c0-559a3b84951a)

Test Timeout:
![image](https://github.com/MarketSquare/robotframework-robocop/assets/8532066/45a81bf1-fe71-4a9c-83c1-08603bba147b)


Sadly without cross referencing it with translation mapping we don't know if "Test timeout" is actually test or task timeout.

Reply:
To sum it up, you can add support to translated section names, but supporting translated setting names is too much work right now. You can add # TODO in the code to denote it could be done in the future.

Reply:
I think we should create an issue for that directly in RF repo. I'll do it to request this change.